source/posix/file_utils.c

```diff
@@ -34,14 +34,17 @@ bool aws_path_exists(const char *path) {
     return stat(path, &buffer) == 0;
 }
 
-int aws_fseek(FILE *file, aws_off_t offset, int whence) {
+int aws_fseek(FILE *file, int64_t offset, int whence) {
 
-    int result =
-#if _FILE_OFFSET_BITS == 64 || _POSIX_C_SOURCE >= 200112L
-        fseeko(file, offset, whence);
+#ifdef AWS_HAVE_POSIX_LARGE_FILE_SUPPORT
+    int result = fseeko(file, offset, whence);
 #else
-        fseek(file, offset, whence);
-#endif
+    /* must use fseek(), which takes offset as a long */
+    if (offset < LONG_MIN || offset > LONG_MAX) {
+        return aws_raise_error(AWS_IO_STREAM_INVALID_SEEK_POSITION);
+    }
+    int result = fseek(file, offset, whence);
```

Comment:
Are there any worries about a relative seek going pass the 32 bit boundary or is that something that will result in an error from fseek itself?

Reply:
I'm going to assume that's handled by fseek. We're just sanitizing the inputs

Reply:
Confirmed, if I do two `fseek(f, LONG_MAX, SEEK_CUR);` in a row the 2nd call fails